File: Python/백준/Silver/5800. 성적 통계/성적 통계.py
import sys
input = sys.stdin.readline

k = int(input())

for i in range(1, k+1, 1):
    data = list(map(int, input().split()))
    n = data[0]
    score = sorted(data[1:], reverse=True)
    largest_gap = max(score[j] - score[j + 1] for j in range(len(score) - 1))

    print(f"Class {i}")
    print(f"Max {score[0]}, Min {score[-1]}, Largest gap {largest_gap}")

File: Java/백준/Silver/5800. 성적 통계/성적 통계.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringBuilder sb = new StringBuilder();

        int k = Integer.parseInt(br.readLine());

        for (int i = 0; i < k; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            int n = Integer.parseInt(st.nextToken());

            int[] score = new int[n];
            for (int j = 0; j < n; j++) {
                score[j] = Integer.parseInt(st.nextToken());
            }

            Arrays.sort(score);
            for (int j = 0; j < score.length / 2; j++) {
                int temp = score[j];
                score[j] = score[score.length - 1 - j];
                score[score.length - 1 - j] = temp;
            }
            int largestGap = 0;
            for (int j = 0; j < score.length - 1; j++) {
                largestGap = Math.max(largestGap, score[j] - score[j + 1]);
            }

            sb.append("Class ").append(i+1).append("\n");
            sb.append("Max ").append(score[0]).append(", Min ").append(score[score.length - 1]).append(", Largest gap ").append(largestGap).append("\n");
        }
        System.out.println(sb);
    }
}